packages/cli/src/lib/errors/snapshotErrors.ts

```diff
@@ -26,24 +32,35 @@ export class SnapshotSynchronizationError extends Error {
       Some conflicts were detected while comparing changes between the snapshot and the target organization.
       Click on the URL below to synchronize your snapshot with your organization before running another push command.
       ${synchronizationPlanUrl}`;
+
+    if (projectPath) {
+      const reportPath = snapshot.saveDetailedReport(projectPath);
+      this.message += dedent`\n\n
+        Detailed report saved at ${reportPath}`;
+    }
   }
 }
 
 export class SnapshotGenericError extends Error {
+  public name = 'Snapshot Error';
   public constructor(
     public snapshot: Snapshot,
     public cfg: Configuration,
-    public pathToReport: string
+    public projectPath?: string
   ) {
     super();
     const report = snapshot.latestReport;
     const urlBuilder = new SnapshotUrlBuilder(cfg);
     const snapshotUrl = urlBuilder.getSnapshotPage(snapshot);
 
     this.message = dedent`Invalid snapshot - ${report.resultCode}.
-      Detailed report saved at ${pathToReport}.
-
       You can also use this link to view the snapshot in the Coveo Admin Console
       ${snapshotUrl}`;
+
+    if (projectPath) {
+      const reportPath = snapshot.saveDetailedReport(projectPath);
+      this.message += dedent`\n\n
+          Detailed report saved at ${reportPath}`;
+    }
```

Comment:
You could refactor this bit, see:
https://www.typescriptlang.org/play?#code/C4TwDgpgBAygdgQzAZwBYHthQLxQXEAbgChRIoBhdOAMwEsBzAVwCcFg7qc8CTi64wCCxoIAxtACSAEQjAEdADYQAJgCUIYdC3kAjZVAgAPIXBXIoAURYttUAN7EoUZIhQZgALlhu0mADROUGC2AFYQYsAACuyoAPzeyMAsAgzEAL7ExDRMcJGccFDJIACCKiqpsvJKqhpaOgAUwrYs3jJyCsrqmtp6ygCUDkF0NFBNNtoAdCHo4ZExwKiDjs7OYtRJUCw9Oguo3M1Trkh+wJPICABuEFWdtTvA4y3TYRHRsf0kq4YTLJMAthBkBcGNAANS4FSqCCCAAGAB04Iigt9vrcaiotg8XFdVHgsAASezbervRbpWFfKCZTLEMSKBDAnwnDwAcRhwjoYmsLUMJhh5isvygdH+YGUgMEFna1S6dV6CH00BWwSY+i5UEQgO4AHJ4CzMEKWjqqWA1YoNes4EkWExItoGijVeqxC5fB5vPr3AEnWaXVAxDQGN4qLRGKx2AVAt8-RbXTM5mT4olkqkgssncgmJAWA1Pk6rZsSb1uMdvWcGUIkvKdFS1hssKxFAAhJhKKEsbhwCAAd2Z5YAqiwW23FB2GoGGPnvoWsGXTkPFNwm6328JJqDgF7TjFQQ15x5807FnRkACgSDoJDoXDJHBLggLZiD4aALRQInFnSTbZZxTAKgoXSSYnWcABNdAmADfA8EUZB0CgJhkGgE8LAtOAAGsigQy46F7IpUGgF8sAEAjoCoa4ELKf5SNDeDlFAj97GIxcKT4b5ijKCo4AYdE5QeBpUP6IIaSAA